libsplinter/src/oauth/rest_api/actix/callback.rs

```diff
@@ -214,6 +264,46 @@ fn new_splinter_access_token() -> String {
     thread_rng().sample_iter(&Alphanumeric).take(32).collect()
 }
 
+/// Gets the user's Biome ID from the session store and saves the user profile information to
+/// the user profile store
+#[cfg(feature = "biome-profile")]
+fn save_user_profile(
+    user_profile_store: Box<dyn UserProfileStore>,
+    oauth_user_session_store: Box<dyn OAuthUserSessionStore>,
+    profile: &OauthProfile,
+    subject: String,
+) -> Result<(), InternalError> {
+    if let Some(user) = oauth_user_session_store
+        .get_user(&subject)
+        .map_err(|err| InternalError::from_source(Box::new(err)))?
+    {
+        let profile = ProfileBuilder::new()
+            .with_user_id(user.user_id().into())
+            .with_subject(profile.subject.clone())
+            .with_name(profile.name.clone())
+            .with_given_name(profile.given_name.clone())
+            .with_family_name(profile.family_name.clone())
+            .with_email(profile.email.clone())
+            .with_picture(profile.picture.clone())
+            .build()
+            .map_err(|err| InternalError::from_source(Box::new(err)))?;
+
+        match user_profile_store.get_profile(&user.user_id()) {
+            Ok(_) => user_profile_store
+                .update_profile(profile)
+                .map_err(|err| InternalError::from_source(Box::new(err))),
+            Err(UserProfileStoreError::InvalidArgument(_)) => user_profile_store
```

Comment:
This can be done in a follow-up PR, but `get_profile` should return `Result<Option<Profile>, UserProfileStoreError>` and `Ok(None)` when the profile doesn't exist. This is generally how we do it because it's more intuitive and idiomatic.